### PR TITLE
[Form] Fixed ChoiceType trim option

### PR DESCRIPTION
**Pull request by HeahDude**
Ref symfony/symfony#26932.

### CODE REVIEW

**Review by javiereguiluz — APPROVED** [review]
Nice! Thanks Jules.

**Comment by HeahDude** [comment]
Thank you for the rewarding 👍

File: reference/forms/types/choice.rst [hunk 1]
```diff
@@ -323,6 +324,14 @@ error_bubbling
 Set that error on this field must be attached to the field instead of
 the parent field (the form in most cases).
 
+trim
+~~~~
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: I think we should move the description to its own file to be able to reuse it in other form types. This is currently missing in some documents (see the failing build).

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Comment by wouterj** [comment]
Ping @HeahDude can you please fix this PR according @xabbuh's comment?

File: reference/forms/types/choice.rst [hunk 2]
```diff
@@ -30,6 +30,7 @@ To use this field, you must specify *either* ``choices`` or ``choice_loader`` op
 | Overridden  | - `compound`_                                                                |
 | options     | - `empty_data`_                                                              |
 |             | - `error_bubbling`_                                                          |
+|             | - `choice_type_trim`_                                                        |
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: here we would still have to use `trim`

Comment (HeahDude) [reply reference/forms/types/choice.rst]: Was not sure ^^. Fixed, thanks!

File: reference/forms/types/options/choice_type_trim.rst.inc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+trim
+~~~~
+
+**type**: ``boolean`` **default**: ``false``
+
+Trimming is disabled by default because the selected value or values must match
+the given choice values exactly (and they could contain white spaces).
```
Comment (xabbuh) [inline reference/forms/types/options/choice_type_trim.rst.inc]: "whitespaces" (without the space)?

**Review by xabbuh — APPROVED** [review]

File: reference/forms/types/options/choice_type_trim.rst.inc (hunk 3 above)
Comment (HeahDude) [reply reference/forms/types/options/choice_type_trim.rst.inc]: Agreed, fixed. Thanks again!

**Comment by wouterj** [comment]
Thanks Jules.

**Pull request merged**